projects/opyn-crab/index.js

```diff
@@ -1,59 +1,22 @@
-const { post } = require('../helper/http')
-const { request, gql } = require("graphql-request");
-const BigNumber = require("bignumber.js");
-
 const ETH = '0x0000000000000000000000000000000000000000';
+const SQEETH = '0xf1b99e3e573a1a9c5e6b2ce818b617f0e664e86b';
+const crabV2 = '0x3B960E47784150F5a63777201ee2B15253D713e8'
 
-// to add TVL under yield category
-async function tvl(timestamp, block, _, { api }){
-    let balances = {};
-
-    const API_URLS = {
-        ethereum: 'https://api.thegraph.com/subgraphs/name/opynfinance/squeeth'
-    };
-    // get eth usd price
-    const key = 'ethereum:0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';      
-    const ethPriceUSD = (await post("https://coins.llama.fi/prices", {
-        "coins": [key]
-    })).coins[key].price;
-    // get eth usd price
-    const squeethKey = 'ethereum:0xf1b99e3e573a1a9c5e6b2ce818b617f0e664e86b';
-    const squeethPriceUSD = (await post("https://coins.llama.fi/prices", {
-        "coins": [squeethKey]
-    })).coins[squeethKey].price;
-
-    // crab strategy vault in squeeth
-    const crabVaultQuery = gql`
-        query Vault($vaultID: ID! = 286) {
-            vault(id: $vaultID) {
-            id
-            shortAmount
-            collateralAmount
-            NftCollateralId
-            owner {
-                id
-            }
-            operator
-            }
-        }
-    `;
-    const crabVaultQueryData = await Promise.all(
-        Object.entries(API_URLS).map(async ([chain, url]) => [
-            chain,
-            (await request(url, crabVaultQuery)).vault,
-        ])
-    );
-
-    const balance = crabVaultQueryData[0][1].collateralAmount - (crabVaultQueryData[0][1].shortAmount * (squeethPriceUSD / ethPriceUSD ));
-    balances = {
-        [ETH]: BigNumber(balances[ETH] || 0).plus(balance).toFixed(0),
-    };
+async function tvl(timestamp, block, _, { api }) {
+  const [_0, _1, ethInCrab, squeethInCrab] = await api.call({
+    target: crabV2,
+    abi: 'function getVaultDetails() view returns (address, uint256, uint256, uint256)'
+  })
 
-    return balances;
+  return {
+    [ETH]: ethInCrab,
+    [SQEETH]: squeethInCrab * -1,
+  };
```

Comment:
The TVL should be the amount of collateral in crab vault in squeeth - (the short amount * squeethEthPrice), so this should be wrong right?

```
    const balance = crabVaultQueryData[0][1].collateralAmount - (crabVaultQueryData[0][1].shortAmount * (squeethPriceUSD / ethPriceUSD ));
 ```

Reply:
I am doing this but instead of fetching squeeth price, I am counting squeeth tokens as negative, if you run the adapter you will get this:
![image](https://user-images.githubusercontent.com/99907941/212572325-882c6d96-cc71-4558-9dce-9b932b50ab72.png)